### PR TITLE
[ENH]: Add support for masks (GM/WM/etc)

### PR DESCRIPTION
**Pull request by fraimondo**
### Are you requiring a new dataset or marker?

- [X] I understand this is not a marker or dataset request

### Which feature do you want to include?

In the same way we support atlases and coordinates, we should be able to support masks to be used, for example, in the spheres masker.

### How do you imagine this integrated in junifer?

Copy/paste the atlases and modify for masks (give or take the family and atlas-specific parameters)

### Do you have a sample code that implements this outside of junifer?

_No response_

### Anything else to say?

a "good" general purpose GM mask :)
https://zenodo.org/record/6463123#.YyHOOrRBzEY

### CODE REVIEW

File: junifer/data/utils.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    resolution: Optional[float],
+    valid_resolution: Union[List[float], List[int], np.ndarray],
+) -> Union[float, int]:
+    """Find the closest resolution.
+
+    Parameters
+    ----------
+    resolution : float
+        The given resolution.
+    valid_resolution : list of float or np.ndarray
```
Comment (synchon) [inline junifer/data/utils.py:19]: list of float or int, or np.ndarray?

File: junifer/markers/crossparcellation_functional_connectivity.py [hunk 2]
```diff
@@ -32,6 +32,9 @@ class CrossParcellationFC(BaseMarker):
     correlation_method : str, optional
         Any method that can be passed to
         :any:`pandas.DataFrame.corr` (default "pearson").
+    mask : str, optional
+        The name of the mask to apply to regions before extracting signals.
+        Check valid options by calling :func:`junifer.data.masks.list_masks`.
```
Comment (synchon) [inline junifer/markers/crossparcellation_functional_connectivity.py:37]: `(default None).`

File: junifer/markers/functional_connectivity_parcels.py [hunk 3]
```diff
@@ -40,6 +40,9 @@ class FunctionalConnectivityParcels(BaseMarker):
     cor_method_params : dict, optional
         Parameters to pass to the correlation function. Check valid options in
         :class:`nilearn.connectome.ConnectivityMeasure` (default None).
+    mask : str, optional
+        The name of the mask to apply to regions before extracting signals.
+        Check valid options by calling :func:`junifer.data.masks.list_masks`.
```
Comment (synchon) [inline junifer/markers/functional_connectivity_parcels.py:45]: `(default None).`

File: junifer/markers/functional_connectivity_spheres.py [hunk 4]
```diff
@@ -43,6 +43,9 @@ class FunctionalConnectivitySpheres(BaseMarker):
     cor_method_params : dict, optional
         Parameters to pass to the correlation function. Check valid options in
         :class:`nilearn.connectome.ConnectivityMeasure` (default None).
+    mask : str, optional
+        The name of the mask to apply to regions before extracting signals.
+        Check valid options by calling :func:`junifer.data.masks.list_masks`.
```
Comment (synchon) [inline junifer/markers/functional_connectivity_spheres.py:48]: `(default None).`

File: junifer/markers/parcel_aggregation.py [hunk 5]
```diff
@@ -35,6 +35,9 @@ class ParcelAggregation(BaseMarker):
     method_params : dict, optional
         Parameters to pass to the aggregation function. Check valid options in
         :func:`junifer.stats.get_aggfunc_by_name`.
+    mask : str, optional
+        The name of the mask to apply to regions before extracting signals.
+        Check valid options by calling :func:`junifer.data.masks.list_masks`.
```
Comment (synchon) [inline junifer/markers/parcel_aggregation.py:40]: `(default None).`

File: junifer/markers/sphere_aggregation.py [hunk 6]
```diff
@@ -37,6 +37,9 @@ class SphereAggregation(BaseMarker):
         (default "mean").
     method_params : dict, optional
         The parameters to pass to the aggregation method (default None).
+    mask : str, optional
+        The name of the mask to apply to regions before extracting signals.
+        Check valid options by calling :func:`junifer.data.masks.list_masks`.
```
Comment (synchon) [inline junifer/markers/sphere_aggregation.py:42]: `(default None).`

File: junifer/data/masks.py [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+_masks_path = Path(__file__).parent / "masks"
+
+"""
+A dictionary containing all supported masks and their respective file or
+data.
+
+The built-in masks are files that are shipped with the package in the
+data/masks directory. The user can also register their own masks.
+"""
+_available_masks: Dict[str, Dict[Any, Any]] = {
```
[the remark this answers is not in the document]
Comment (fraimondo) [reply junifer/data/masks.py:28]: not entirely. Inner values could be anything

[the remark this answers is not in the document]
Comment (synchon) [reply junifer/data/masks.py]: But shouldn't be the inner dict keys be string?

[the remark this answers is not in the document]
Comment (synchon) [reply junifer/data/masks.py]: Okay now, saw the change.

File: junifer/data/tests/test_data_utils.py [hunk 8]
```diff
@@ -26,7 +26,17 @@
...
     resolution: float, valid_resolutions: List[float], expected: float
 ):
-    """Test closest_resolution."""
+    """Test closest_resolution.
+
+    Parameters
+    ----------
+    resolution: float
+        The resolution to test.
+    valid_resolutions: List[float]
```
Comment (synchon) [inline junifer/data/tests/test_data_utils.py:35]: `list of float`

**Review by synchon — CHANGES_REQUESTED** [review]

**Comment by synchon** [comment]
Will wait for the CI to complete before merging.

**Pull request merged**